### PR TITLE
Add native YAML export

### LINKED ISSUE #161: yaml.dump of spec.to_dict() results in invalid spec
I'm trying to dump the spec to YAML with `yaml.dump(spec.to_dict())` but that results in an invalid representation due to entries like these:

```yaml
  - - /api/v0/groups
    - !!python/object/new:apispec.core.Path
```

Or, for items:
```yaml
items:
                  properties: !!python/object/new:apispec.lazy_dict.LazyDict
```
```yaml
403:
              description: You are not authorized to request this resource
              schema:
                properties: !!python/object/new:apispec.lazy_dict.LazyDict
                  dictitems:
                    code: {format: int32, type: integer}
                    error: {type: string}
                required: [code, error]
                type: object
```

In order to dump this correctly it seems we'd need a custom YAML dumper for apispec. Use `safe_dumper` isn't possible either since it doesn't know how to dump certain of these objects.

### PR DESCRIPTION
**Pull request by jd**
Fixes #161

### CODE REVIEW

**Comment by jd** [comment]
Fix flake8 on Python 3

**Comment by sloria** [comment]
Thanks @jd ! This is looking good. Just a few more things before this is merged:

- Get branch up to date with `dev`
- Update documentation 
- Add yourself to AUTHORS.rst

**Comment by jd** [comment]
Update doc, README and rebased on dev

File: apispec/core.py [hunk 1]
```diff
@@ -387,3 +393,26 @@ def validate_openapi_version(openapi_version_str):
         )
 
     return openapi_version
+
+
+class YAMLDumper(yaml.Dumper):
+    pass
```
Comment (sloria) [inline apispec/core.py]: Remove this.

File: docs/quickstart.rst [hunk 2]
```diff
@@ -66,6 +66,25 @@ To output your OpenAPI spec, invoke the `to_dict <apispec.APISpec.to_dict>` meth
     # 'swagger': '2.0',
     # 'tags': []}
 
+If you want to export your API in YAML, for example to feed a third-party tool,
```
Comment (sloria) [inline docs/quickstart.rst]: Proposed revision:

```
Use to_yaml to export your spec to YAML.
```

File: docs/special_topics.rst [hunk 3]
```diff
@@ -27,26 +27,30 @@ To add additional fields (e.g. ``"discriminator"``) to Schema objects generated
 Rendering to YAML or JSON
 -------------------------
 
-`apispec` will only serialize your spec to a Python dictionary. Use an appropriate module to render your spec to YAML or JSON.
+`apispec` can serialize your spec to a Python dictionary or to YAML.
```
Comment (sloria) [inline docs/special_topics.rst]: I think this sentence can be removed.

File: apispec/core.py [hunk 4]
```diff
@@ -387,3 +393,26 @@ def validate_openapi_version(openapi_version_str):
...
+
+    @staticmethod
+    def _represent_dict(dumper, instance):
+        return dumper.represent_mapping('tag:yaml.org,2002:map',
+                                        instance.items())
+
+    if PY2:
+        def _unicode_representer(dumper, uni):
```
Comment (sloria) [inline apispec/core.py]: This should probably be `_represent_unicode` for consistency.

File: apispec/core.py [hunk 5]
```diff
@@ -387,3 +393,26 @@ def validate_openapi_version(openapi_version_str):
         )
 
     return openapi_version
+
+
+class YAMLDumper(yaml.Dumper):
+    pass
+
+    @staticmethod
```
Comment (sloria) [inline apispec/core.py]: Does this need to be a staticmethod? If so, it should be consistent with the unicode representer method.

**Review by sloria — CHANGES_REQUESTED** [review]
Thanks for updating the docs. Just a few minor changes suggested.

File: apispec/core.py (hunk 1 above)
Comment (jd) [reply apispec/core.py]: Done.

File: apispec/core.py (hunk 4 above)
Comment (jd) [reply apispec/core.py]: Done.

File: apispec/core.py (hunk 5 above)
Comment (jd) [reply apispec/core.py]: Done.

File: docs/quickstart.rst (hunk 2 above)
Comment (jd) [reply docs/quickstart.rst]: Done.

File: docs/special_topics.rst (hunk 3 above)
Comment (jd) [reply docs/special_topics.rst]: Done.

**Comment by jd** [comment]
Pull-request updated, HEAD is now 83c940e6c8ca782e6542f0ed1035560291fd008a

**Review by sloria — APPROVED** [review]

**Comment by sloria** [comment]
Thanks!

**Pull request merged**